app/views/core/AnonymousTeacherModal.coffee

```diff
@@ -46,11 +57,23 @@ module.exports = class AnonymousTeacherModal extends ModalView
     else
       @state.set 'checkEmailState', 'invalid'
 
+  onChangeAnonymousStudentNameInput: (e) ->
+    name = @getStudentName()
+    if not name
+      @state.set 'checkNameState', 'none'
+    else if not _.isEmpty(name)
+      @state.set 'checkNameState', 'valid'
+    else
+      @state.set 'checkNameState', 'invalid'
+
   onClickAnonymousTeacherEmailSendButton: (e) ->
+    return if @state.get 'sendEmailState' is 'sent'
     @state.set 'sendEmailState', 'sending'
-    contact.sendTeacherSignupInstructions(@getEmail())
+    contact.sendTeacherSignupInstructions(@getEmail(), @getStudentName())
       .then =>
         @state.set 'sendEmailState', 'sent'
+        storage.save('teacher signup email sent', true)
```

Comment:
I won't block on this but I want to point out that this is more of that code smell I mentioned a week or two ago about being wary of using.

Reply:
Yup, how would you improve this?